src/main/java/duke/dukehelper/CommandHandler.java

```diff
@@ -0,0 +1,364 @@
+/**
+ * Class to handle command inputs by user
+ */
+package duke.dukehelper;
+import duke.taskhelper.ListTasks;
+import duke.taskhelper.Packet;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import static duke.dukehelper.Constants.Error.NO_ERROR;
+import static duke.dukehelper.Routine.replyRoutine;
+
+public class CommandHandler {
+    public static ListTasks list = new ListTasks();
+    private static SaveManager saveManager = null;
+    private static SaveManager loadManager = null;
+    private static int conversationCounter = 0;
+    /**
+     * Validates if the input supplies a description as needed by the command
+     * @param packet
+     * @throws DukeException.InvalidDescription
+     */
+    private static void validatePayload(Packet packet) throws DukeException.InvalidDescription {
+        if (packet.getPacketPayload() == null){
+            throw new DukeException.InvalidDescription();
+        }
+    }
+
+    /**
+     * Validates if list is empty
+     * @throws DukeException.NoList
+     */
+    private static void validateList() throws DukeException.NoList {
+        if (list.getNumTasks() == 0){
+            throw new DukeException.NoList();
+        }
+    }
+    /**
+     * Handles command and prints messages, if any onto console depending on input parameter.
+     *
+     * @param command a string of a simple root word to represent a particular message
+     * @param packet an input parameter for the message to print, given the command requires further inputs.
+     */
+    public static Constants.Command handleCommand(Constants.Command command, Packet packet, boolean verbose, boolean isReply){
+        String output = "";
+        String customErrorMessage = "";
+        boolean isDrawPartition = true;
+        // If true, data will not be overwritten, and user will be prompted to input to respond to the command requirements.
+        boolean exit = false;
+        // If a command demands a response from the user, this variable will relay to Duke the command to reply to.
+        // If no reply is needed, this variable stays at null.
+        Constants.Command commandToReply = null;
+        // Error code where applicable.
+        Constants.Error err = null;
+
+        switch(command){
+        case HELLO:
```

Comment:
I really love the use of enums here. Just looking at simple English words in all CAPS in a switch-case really makes my day.

Reply:
Thank you! It is my life goal to make exceedingly readable switch cases and your words give my life meaning.